src/java.base/share/classes/java/time/chrono/ThaiBuddhistDate.java

```diff
@@ -315,25 +315,23 @@ public ThaiBuddhistDate with(TemporalField field, long newValue) {
             if (getLong(chronoField) == newValue) {
                 return this;
             }
-            switch (chronoField) {
-                case PROLEPTIC_MONTH:
+            return switch (chronoField) {
+                case PROLEPTIC_MONTH -> {
                     getChronology().range(chronoField).checkValidValue(newValue, chronoField);
-                    return plusMonths(newValue - getProlepticMonth());
-                case YEAR_OF_ERA:
-                case YEAR:
-                case ERA: {
+                    yield  plusMonths(newValue - getProlepticMonth());
+                }
+                case YEAR_OF_ERA -> {
                     int nvalue = getChronology().range(chronoField).checkValidIntValue(newValue, chronoField);
-                    switch (chronoField) {
-                        case YEAR_OF_ERA:
-                            return with(isoDate.withYear((getProlepticYear() >= 1 ? nvalue : 1 - nvalue)  - YEARS_DIFFERENCE));
-                        case YEAR:
-                            return with(isoDate.withYear(nvalue - YEARS_DIFFERENCE));
-                        case ERA:
-                            return with(isoDate.withYear((1 - getProlepticYear()) - YEARS_DIFFERENCE));
-                    }
+                    yield with(isoDate.withYear((getProlepticYear() >= 1 ? nvalue : 1 - nvalue) - YEARS_DIFFERENCE));
                 }
-            }
-            return with(isoDate.with(field, newValue));
+                case YEAR -> {
+                    int nvalue = getChronology().range(chronoField).checkValidIntValue(newValue, chronoField);
+                    yield with(isoDate.withYear(nvalue - YEARS_DIFFERENCE));
+                }
+                case ERA -> with(isoDate.withYear((1 - getProlepticYear()) - YEARS_DIFFERENCE));
```

Comment:
`checkValidIntValue` performs validation, so removing it has changed the behavoiur

Reply:
Changes reverted. See 2ae4a57